src/ale_interface.cpp

```diff
@@ -123,15 +178,44 @@ void ALEInterface::loadROM(string rom_file = "") {
     rom_file = theOSystem->romFile();
   }
   loadSettings(rom_file, theOSystem);
-  romSettings.reset(buildRomRLWrapper(rom_file));
+
+  RomSettings* romRlWrapper = buildRomRLWrapper(rom_file);
+  if (romRlWrapper == NULL) {
+    Logger::Info << "Unable to map rom '" << rom_file
+        << "'. Trying alternative method..." << std::endl;
+
+    std::string s = "Unknown";
+    std::string s1 = s.substr(0, s.find("("));
+    s1.erase(std::remove_if(s1.begin(), s1.end(), is_illegal), s1.end());
+    std::cout << s1 << std::endl;
+
+    const Properties properties = theOSystem->console().properties();
+    const std::string md5 = properties.get(Cartridge_MD5); //TODO there shouldn't be a need to do this twice
+    string rom_candidate = find_rom(md5);
+    if (rom_candidate != "") {
```

Comment:
Hm. This code is not really necessary for the md5 stuff; it just happened that I was working in the vicinity and I got segfaults when using roms from the wild that didn't match the naming convention.

It's probably best to reduce this PR to just the MD5 part, then we can discuss how to handle "unexpected ROM" that goes beyond using the correct name with an "unsupported" ROM there.

Raising exception would just be a slightly friendlier version than the segfault, and it would force users to rename all there ROMs for no reason that I can think of: The ROMs can all be identified via their MD5 checksums, it should be completely irrelevant what the filenames are. AFAIR the previous method even had issues when using ROMs outside the current directory, although that may have been local to my own modified version.